JsonLedgerApiSample/src/main/java/com/example/services/ScanProxy.java

```diff
@@ -37,4 +41,18 @@ public String getDsoPartyId() throws ApiException {
         GetDsoPartyIdResponse response = this.scanProxyApi.getDsoPartyId();
         return response.getDsoPartyId();
     }
+
+    public Optional<ContractWithState> getTransferPreapproval(String partyId) throws ApiException {
```

Comment:
This is the CC specific variant to figure out whether a transfer will be executed directly.

The generic option would be to attempt to initiate a transfer to the receiver and check whether the registry sets `transfer_kind` to `direct`: https://github.com/hyperledger-labs/splice/blob/7c393f826c73843f9215bf960b5da267ababc5fa/token-standard/splice-api-token-transfer-instruction-v1/openapi/transfer-instruction-v1.yaml#L192-L204

So the question is: where do we want to steer our integrators?